lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -550,200 +425,246 @@ export class UserAgentApplication {
-      this.cacheStorage.setItem(Constants.msalErrorDescription, ClientAuthErrorMessage.endpointResolutionError.desc);
+      // set state in cache
+      if (interactionType === Constants.interactionTypeRedirect) {
+        if (!isLoginCall) {
+          this.cacheStorage.setItem(Constants.stateAcquireToken, serverAuthenticationRequest.state, this.inCookie);
+        }
+      } else {
+        window.renewStates.push(serverAuthenticationRequest.state);
+        if (isLoginCall) {
+          window.requestType = Constants.login;
+        } else {
+          window.requestType = Constants.renewToken;
+        }
 
-      // reject that is passed in - REDO this in the subsequent refactor, passing reject is confusing
-      if (reject) {
-        reject(ClientAuthError.createEndpointResolutionError());
+        // Register callback to capture results from server
+        this.registerCallback(serverAuthenticationRequest.state, scope, resolve, reject);
       }
 
-      // Close the popup window
+      // prompt user for interaction
+      this.navigateWindow(urlNavigate, popUpWindow);
+    }).catch((err) => {
+      this.logger.warning("could not resolve endpoints");
+      this.errorHandler(interactionType, ClientAuthError.createEndpointResolutionError(err.toString), buildResponseStateOnly(request.state), reject);
       if (popUpWindow) {
         popUpWindow.close();
       }
-    // this is an all catch for any failure for the above code except the specific 'reject' call
-    }).catch((err) => {
-      this.logger.warning("could not resolve endpoints");
-      reject(ClientAuthError.createEndpointResolutionError(err.toString));
     });
   }
 
   /**
-   * Use when you want to obtain an access_token for your API via opening a popup window in the user's browser
+   * Use this function to obtain a token before every call to the API / resource provider
+   *
+   * MSAL return's a cached token when available
+   * Or it send's a request to the STS to obtain a new token using a hidden iframe.
+   *
    * @param {@link AuthenticationParameters}
    *
    * To renew idToken, please pass clientId as the only scope in the Authentication Parameters
    * @returns {Promise.<AuthResponse>} - a promise that is fulfilled when this function has completed, or rejected if an error was raised. Returns the {@link AuthResponse} object
+   *
    */
-  acquireTokenPopup(request: AuthenticationParameters): Promise<AuthResponse> {
+  @resolveTokenOnlyIfOutOfIframe
+  acquireTokenSilent(request: AuthenticationParameters): Promise<AuthResponse> {
     return new Promise<AuthResponse>((resolve, reject) => {
+
       // Validate and filter scopes (the validate function will throw if validation fails)
       this.validateInputScope(request.scopes, true);
 
       const scope = request.scopes.join(" ").toLowerCase();
 
-      // Get the account object if a session exists
+      // if the developer passes an account give him the priority
```

Comment:
I'd reword this to something like "Passed in account should take precedence"

Reply:
done